packages/react/menu/src/MenuTypeahead.tsx

```diff
@@ -0,0 +1,189 @@
+import * as React from 'react';
+import { createContext } from '@radix-ui/react-context';
+import { createCollection } from '@radix-ui/react-collection';
+import { composeEventHandlers } from '@radix-ui/primitive';
+import { composeRefs } from '@radix-ui/react-compose-refs';
+import { Primitive } from '@radix-ui/react-primitive';
+
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
+
+type ItemElement = React.ElementRef<typeof MenuTypeaheadItem>;
+
+const [CollectionSlot, CollectionItemSlot, useCollection] = createCollection<ItemElement, {}>();
```

Comment:
There is some overlap here as `Menu` does already handle a collection but I found having this makes the separation much cleaner.

Reply:
I know we're potentially getting rid of the separate component idea but just wanted to say I agree with this decision haha, I'd have done the same :+1: